### ISSUE TITLE
Clash between defaultParams and comma-spacing

### ISSUE THREAD

**Issue by davidtheclark** [eslint/eslint#1845]
If I have an argument with a default param and another argument after it, comma-spacing throws an inaccudate error. Here's a simple example:

test.json
```json
{
	"ecmaFeatures": {
		"defaultParams": true
	},
	"rules": {
		"comma-spacing": 2
	}
}
```

test.js
```js
'use strict';

function foo(bar = 1, baz = 2) {
	return bar + baz;
}

foo();
```

CLI
```bash
 % eslint --c test.json test.js

test.js
  3:22  error  There should be no space before ','  comma-spacing

✖ 1 problem (1 error, 0 warnings)
```

**Comment by gyandeeps** [comment]
I dont think this scenarios is accounted for. Possibly a bug.

**Comment by nzakas** [comment]
Yeah, comma spacing is a beast of a feature, I'm not surprised that we'd run into an issue with default parameters.
